docs/internals/generic/_index.md

```diff
@@ -0,0 +1,215 @@
+Model | Description
+----- | -----------
+[`fake`](fake/) | A model used for testing a generic service.
+
+### Modular resources
+
+{{<modular-resources api="rdk:service:generic" type="generic">}}
+
+## Control your machine with Viam's client SDK libraries
+
+To get started using Viam's SDKs to connect to and control your machine, go to your machine's page on [the Viam app](https://app.viam.com), navigate to the **Code sample** tab, select your preferred programming language, and copy the sample code generated.
+
+{{% snippet "show-secret.md" %}}
+
+When executed, this sample code will create a connection to your machine as a client.
+Then control your machine programmatically by getting your `generic` service from the machine with `FromRobot` and adding API method calls, as shown in the following examples.
+
+Be sure to import the generic service package for the SDK you are using:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```python
+from viam.services.generic import Generic
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go
+import (
+  "go.viam.com/rdk/services/generic"
+)
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+```cpp
+#include <viam/sdk/services/generic/generic.hpp>
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## API
+
+The generic service supports the following method:
+
+{{< readfile "/static/include/services/apis/generic.md" >}}
+
+### DoCommand
+
+Execute model-specific commands.
+If you are implementing your own generic service and add features that have no built-in API method, you can access them with `DoCommand`.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
+
+- `command` [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): The command to execute.
+
+**Returns:**
+
+- [(Dict[str, Any])](https://docs.python.org/3/library/stdtypes.html#typesmapping): Result of the executed command.
+
+```python {class="line-numbers linkable-line-numbers"}
+my_generic = Generic.from_robot(robot=robot, name="my_generic_service")
+
+raw_dict = {
+  "command": "raw",
+  "raw_input": "home"
+}
+await my_generic.do_command(raw_dict)
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/generic/client/index.html#viam.services.generic.client.GenericClient.do_command).
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+**Parameters:**
+
+- `ctx` [(Context)](https://pkg.go.dev/context): A Context carries a deadline, a cancellation signal, and other values across API boundaries.
+- `cmd` [(map[string]interface{})](https://go.dev/blog/maps): The command to execute.
+
+**Returns:**
+
+- [(map[string]interface{})](https://go.dev/blog/maps): Result of the executed command.
+- [(error)](https://pkg.go.dev/builtin#error): An error, if one occurred.
+
+```go {class="line-numbers linkable-line-numbers"}
+myGeneric, err := generic.FromRobot(robot, "my_generic_service")
+
+resp, err := myGeneric.DoCommand(ctx, map[string]interface{}{"command": "example"})
+```
```

Comment:
Go code for using generic service appears identical to that for generic component. Please correct if I am wrong!

Reply:
Looks right to me!